packages/core/src/odata-common/entity-builder.ts

```diff
@@ -1,12 +1,28 @@
-import { createLogger } from '@sap-cloud-sdk/util';
+import {
+  createLogger,
+  partition,
+  upperCaseSnakeCase
+} from '@sap-cloud-sdk/util';
 import type { Constructable, Entity } from './entity';
-import { toStaticPropertyFormat } from './name-converter';
+import { isNavigationProperty } from './properties-util';
 
 const logger = createLogger({
   package: 'core',
   messageContext: 'entity-builder'
 });
 
+type FromJsonType<T> = {
```

Comment:
could you add some explaining text which this type does. My conditional type magic is not so powerful to grasp this quicly.

Reply:
Done.